src/Raven.Server/Documents/Handlers/Debugging/QueriesDebugHandler.cs

```diff
@@ -93,5 +95,112 @@ public Task RunningQueries()
 
             return Task.CompletedTask;
         }
+       
+        [RavenAction("/databases/*/debug/queries/cache/list", "GET", AuthorizationStatus.ValidUser, IsDebugInformationEndpoint = true)]
+        public Task QueriesCacheList()
+        {
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                var queryCache = Database.QueryMetadataCache.GetQueryCache();
+
+                var responseDjv = new DynamicJsonValue();
+                var queriesList = new DynamicJsonArray();
+
+                responseDjv["CachedQueryMetadataItems"] = queriesList;
+
+                foreach (var item in queryCache)
+                {   
+                    if (item != null)
+                    {
+                        var curDjvItem = new DynamicJsonValue();
+                        queriesList.Add(curDjvItem);
+
+                        if (item.IsGroupBy)
+                        {
+                            curDjvItem["IsGroupBy"] = true;                            
```

Comment:
Let's add .ToJson() method inside QueryMetadata and handle that there. Remember to use `nameof`.

Reply:
I'm not sure that ToJson is appropriate here, because I don't think that we need "all" the cache properties, we want to minimize it to the most important info to the user, that's why I don't "always" write the properties, only when it's interesting

Reply:
You can apply same logic there

Reply:
And then ToJson won't be the ToJson that we know as the function for full serialization. I think that the convention in QueriesDebugHandler right now is to create the serialization "by hand", I think that we better be consistent

Reply:
Use `nameof` then

Reply:
Yes, it is important